### PR TITLE
Dalec E2E: Use upstream k8s images if no artifacts are built

### PR DESCRIPTION
**Pull request by willie-yao**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR accounts for an edge case where a Dalec test is only testing images and a CAPZ community gallery image does not yet exist for that version, the artifacts are set to the wrong version.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] cherry-pick candidate

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: templates/test/ci/prow-dalec-custom-builds/patches/azl3-machine-deployment.yaml [hunk 1]
```diff
@@ -110,7 +110,45 @@ spec:
...
+            # All downloads succeeded — now stop kubelet and replace binaries
+            systemctl stop kubelet
+            for BINARY in "$${ALL_UPSTREAM_BINARIES[@]}"; do
+              mv "/tmp/$${BINARY}.download" "/usr/bin/$${BINARY}"
+            done
+
+            # Clean up stale kubelet flags in /etc/sysconfig/kubelet.
+            # The gallery image may ship flags removed in newer k8s versions
+            # (e.g. --pod-infra-container-image was removed in v1.35).
+            if [ -f /etc/sysconfig/kubelet ]; then
```
Comment (mboersma) [inline templates/test/ci/prow-dalec-custom-builds/patches/azl3-machine-deployment.yaml:143]: Do we only need this "sanitize" block for azl3, not the other flavors?

File: templates/test/ci/prow-dalec-custom-builds/patches/azl3-machine-deployment.yaml [hunk 2]
```diff
@@ -110,7 +110,45 @@ spec:
...
+            echo "============================================================"
+            echo "DALEC IMAGE-ONLY TEST: No dalec binary revisions set."
+            echo "Downloading official upstream binaries ($${TARGET_VERSION})"
+            echo "from dl.k8s.io for version consistency with dalec images."
+            echo "============================================================"
+
+            # Download all binaries to temp files first, then replace
+            ALL_UPSTREAM_BINARIES=("kubeadm" "kubectl" "kubelet")
+            for BINARY in "$${ALL_UPSTREAM_BINARIES[@]}"; do
+              TEMP_FILE="/tmp/$${BINARY}.download"
```
Comment (mboersma) [inline templates/test/ci/prow-dalec-custom-builds/patches/azl3-machine-deployment.yaml]: It doesn't really matter in this context, but `mktemp` can be more portable than hardcoding `/tmp`.

**Review by mboersma — COMMENTED** [review]
lgtm, my only real concern is it contains 30 lines of code that is duplicated twice (with one variant). Is it possible to have just one patch file?

File: templates/test/ci/prow-dalec-custom-builds/patches/azl3-machine-deployment.yaml (hunk 1 above)
Comment (willie-yao) [reply templates/test/ci/prow-dalec-custom-builds/patches/azl3-machine-deployment.yaml:143]: There was a sysconfig issue that was specific to azl3. Ubuntu uses etc/default/kubelet

**Comment by willie-yao** [comment]
> my only real concern is it contains 30 lines of code that is duplicated twice (with one variant). Is it possible to have just one patch file?

Some of the scripts differ very slightly between distros and control planes / machine deployments have different paths, so I don't think it's trivial to do this with just a single patch file. It could be something to look into as a follow-up though!

**Review by mboersma — APPROVED** [review]
/lgtm
/approve
